src/pkg/cli/tail.go

```diff
@@ -265,6 +279,12 @@ func Tail(ctx context.Context, client client.Client, service, etag string, since
 					line = pkg.StripAnsi(line)
 				}
 				fmt.Println(line)
+
+				// Detect end logging event
+				if params.EndEventDetectFunc != nil && params.EndEventDetectFunc(msg.Service, msg.Host, line) {
+					cancel()
```

Comment:
wonder if we should just have a `defer cancel()` right after the line `WithCancel`, this (and the other one) can simply `return nil` and be done